akka-actor-typed/src/main/scala/akka/actor/typed/internal/receptionist/ReceptionistImpl.scala

```diff
@@ -148,6 +154,44 @@ private[akka] object ReceptionistImpl extends ReceptionistBehaviorProvider {
           externalInterface.onUnregister(key, serviceInstance)
           updateRegistry(Set(key), _.removed(key)(serviceInstance))
 
+        case NodesRemoved(addresses) ⇒
+          if (addresses.isEmpty)
+            Behaviors.same
+          else {
+            import akka.actor.typed.scaladsl.adapter._
+            val localAddress = ctx.system.toUntyped.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+
+            def isOnRemovedNode(ref: ActorRef[_]): Boolean = {
+              if (ref.path.address.hasLocalScope) addresses(localAddress)
```

Comment:
What does that do exactly? When do we expect our own node to be reported as removed?